src/thor/triplegbuilder.cc

```diff
@@ -1670,6 +1674,13 @@ void TripLegBuilder::Build(
       }
     }
 
+    // Set length if requested. Convert to km
+    if (controller.attributes.at(kEdgeLength)) {
```

Comment:
and finally we set the length once we are sure how much of it we will use after trimming